packages/api/src/controllers/stripe.ts

```diff
@@ -214,6 +225,70 @@ const sendUsageRecordToStripe = async (
   );
 };
 
+export const calculateOverageOnMinimumSpend = async (product, billingUsage) => {
+  const minimumSpend = product.monthlyPrice;
+
+  let storageUsage = billingUsage.storageUsage;
+  let deliveryUsage = billingUsage.deliveryUsage;
+  let totalUsage = billingUsage.totalUsage;
+
+  let prices: any = {};
+
+  if (product?.usage) {
+    product.usage.forEach((item) => {
+      if (item.name.toLowerCase() === "transcoding")
+        prices.transcoding = item.price;
+      if (item.name.toLowerCase() === "delivery") prices.streaming = item.price;
+      if (item.name.toLowerCase() === "storage") prices.storage = item.price;
+    });
+  }
+
+  let totalSpent =
+    prices.storage * storageUsage +
+    prices.streaming * deliveryUsage +
+    prices.transcoding * totalUsage;
+  if (totalSpent > minimumSpend) {
+    let remainingOverage = totalSpent - minimumSpend;
+    let overageUsage = {
+      StorageUsageMins: 0,
+      DeliveryUsageMins: 0,
+      TotalUsageMins: 0,
+    };
+
+    if (remainingOverage > 0) {
+      const totalSpentInCategories = {
+        storage: prices.storage * storageUsage,
+        delivery: prices.streaming * deliveryUsage,
+        transcoding: prices.transcoding * totalUsage,
+      };
+
+      const totalSpent =
+        totalSpentInCategories.storage +
+        totalSpentInCategories.delivery +
+        totalSpentInCategories.transcoding;
+
+      if (totalSpent > 0) {
+        const overageRatio = remainingOverage / totalSpent;
+
+        overageUsage.StorageUsageMins =
+          (totalSpentInCategories.storage * overageRatio) / prices.storage;
+        overageUsage.DeliveryUsageMins =
+          (totalSpentInCategories.delivery * overageRatio) / prices.streaming;
+        overageUsage.TotalUsageMins =
+          (totalSpentInCategories.transcoding * overageRatio) /
+          prices.transcoding;
+      }
+    }
+
+    return overageUsage;
+  }
+  return {
```

Comment:
**nit:** A general tip to keep less indentation in the code is also to just invert some conditions. e.g. in this case instead of 
```
if (total > minimum) {
  /* a long branch of code */
}
// POV: You're reading the code linearly... "where was I again?" 😆 
return 0
```
it could be
```
if (total <= minimum) {
  return 0
}
/* the long branch of code, with 1 level less indentation 😌  */
```